packages/server/app.ts

```diff
@@ -433,46 +443,113 @@ export async function init() {
   // Should perhaps be done manually?
   await migrateDbToLatest({ region: 'main', db: knex })
 
-  app.use(cookieParser())
-  app.use(DetermineRequestIdMiddleware)
-  app.use(initiateRequestContextMiddleware)
-  app.use(determineClientIpAddressMiddleware)
-  app.use(LoggingExpressMiddleware)
-
+  app.use(
+    handleErrors({
+      handler: cookieParser(),
+      verbPhraseForErrorMessage: 'parsing cookies',
+      defaultErrorType: CookieParserError
+    })
+  )
+  app.use(
+    handleErrors({
+      handler: DetermineRequestIdMiddleware,
+      verbPhraseForErrorMessage: 'determining request id'
+    })
+  )
+  app.use(
+    handleErrors({
+      handler: initiateRequestContextMiddleware,
+      verbPhraseForErrorMessage: 'initiating request context'
+    })
+  )
   if (asyncRequestContextEnabled()) {
     startupLogger.info('Async request context tracking enabled 👀')
   }
 
-  if (process.env.COMPRESSION) {
-    app.use(compression())
+  app.use(
+    handleErrors({
```

Comment:
instead of having to do all this for every third party middleware, why not just add an extra middleware at the very end of the chain, before defaultErrorHandler & errorMetricsMiddleware, that formats the error in a way that the subsequent error tracking/handling middlewares can handle it?

Reply:
i see that there's custom options for each usage, but it's just a bunch of extra complexity that I'm not sure is worth it. i'd rather have them all format to the same error subtype and be able to forget about this, rather than have to remember to add this wrapper for all future third party middlewares